data/src/main/java/org/apache/iceberg/data/TableMigrationUtil.java

```diff
@@ -37,16 +37,16 @@
 import org.apache.iceberg.PartitionField;
 import org.apache.iceberg.PartitionSpec;
 import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.exceptions.RuntimeIOException;
 import org.apache.iceberg.hadoop.HadoopInputFile;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.mapping.NameMapping;
 import org.apache.iceberg.orc.OrcMetrics;
 import org.apache.iceberg.parquet.ParquetUtil;
-import org.apache.parquet.hadoop.ParquetFileReader;
-import org.apache.parquet.hadoop.metadata.ParquetMetadata;
+import org.apache.iceberg.util.Tasks;
 
 public class TableMigrationUtil {
-
+  public static final String PARQUET_READ_PARALLELISM = "parquet.metadata.read.parallelism";
```

Comment:
Seems odd to me that we pass this through as a Hadoop Configuration, I feel like it should be a method parameter of listPartition? Otherwise implementations will be passing through this information by modifying the hadoop conf which seems a bit odd to me

Reply:
I complelely agree with your point .   Maybe rewrite like this ?
```java
  @Deprecated
  public static List<DataFile> listPartition(Map<String, String> partition, String uri, String format,
                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig,
                                             NameMapping mapping) {
    int parallelism = 1;
    if (format.contains("avro")) {
      return listAvroPartition(partition, uri, spec, conf, parallelism);
    } else if (format.contains("parquet")) {
      return listParquetPartition(partition, uri, spec, conf, metricsConfig, mapping, parallelism);
    } else if (format.contains("orc")) {
      return listOrcPartition(partition, uri, spec, conf, metricsConfig, mapping, parallelism);
    } else {
      throw new UnsupportedOperationException("Unknown partition format: " + format);
    }
  }

  public static List<DataFile> listPartition(Map<String, String> partition, String uri, String format,
                                             PartitionSpec spec, Configuration conf, MetricsConfig metricsConfig,
                                             NameMapping mapping, int parallelism) {
    if (format.contains("avro")) {
      return listAvroPartition(partition, uri, spec, conf, parallelism);
    } else if (format.contains("parquet")) {
      return listParquetPartition(partition, uri, spec, conf, metricsConfig, mapping, parallelism);
    } else if (format.contains("orc")) {
      return listOrcPartition(partition, uri, spec, conf, metricsConfig, mapping, parallelism);
    } else {
      throw new UnsupportedOperationException("Unknown partition format: " + format);
    }
  }
```